src/movement/maestro.hpp

```diff
@@ -0,0 +1,342 @@
+/**
+ * @author Ryan Summers
+ * @date 10-25-2016
+ *
+ * @brief Provides declaration and definitions of the thruster controller
+ *        class.
+ *
+ * @note Any documentation specified in this file refers to information located
+ *       at https://www.pololu.com/docs/0J40. Information is formatted into
+ *       section and subsection. For example, (4.5) refers to Section 4
+ *       subsection 5.
+ */
+
+#ifndef THRUSTER_CONTROLLER_H
+#define THRUSTER_CONTROLLER_H
+
+#include "utility/serial.hpp"
+#include <ros/ros.h>
+#include <vector>
+
+using std::vector;
+
+namespace rs {
+    /**
+     * Thruster controlling class.
+     *
+     * @brief This class provides an interface for communicating with the
+     *        Pololu Maestro servo controller. These servo commands are then
+     *        translated into a thruster power by the ESC (electronic speed
+     *        controller).
+     *
+     * @tparam thrusters The number of thrusters to be controlled by the class.
+     * @tparam max_speed The software limit on the maximum speed that any
+     *         thruster may be set to.
+     */
+    class ThrusterController
+    {
+        /*
+         * Specifies the delay in seconds after which a reset signal (1500
+         * microsecond pulse) should be sent.
+         */
+        static constexpr double reset_delay = 120.0;
+
```

Comment:
It was originally templated, which is why all the code is here. Do we want to cut it into two files?